File: archetype-base/src/test/java/com/kelin/archetype/base/utils/RsaUtilTest.java
// Copyright 2020 Kelin Inc. All rights reserved.

package com.kelin.archetype.base.utils;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

import java.security.KeyPair;

/**
 * @author Kelin Tan
 */
public class RsaUtilTest {
    @Test
    public void testRsa() throws Exception {
        String src = "test";

        KeyPair keyPair = RsaUtil.generatorKeyPair();

        String privateKey = RsaUtil.encodePrivateKey(keyPair);
        String publicKey = RsaUtil.encodePublicKey(keyPair);

        String encrypt = RsaUtil.encrypt(src, publicKey);

        String decrypt = RsaUtil.decrypt(encrypt, privateKey);

        assertEquals(decrypt, src);
    }
}

File: archetype-jwt/src/test/java/com/kelin/archetype/jwt/RsaUtilTest.kt
// Copyright 2020 Kelin Inc. All rights reserved.

package com.kelin.archetype.jwt

import com.kelin.archetype.test.KtTestUtils
import org.junit.Test

/**
 * @author Kelin Tan
 */
class RsaUtilTest : KtTestUtils {
    @Test
    @Throws(Exception::class)
    fun testRsa() {
        val src = "test"
        val keyPair = RsaUtil.generatorKeyPair()
        val privateKey = RsaUtil.encodePrivateKey(keyPair)
        val publicKey = RsaUtil.encodePublicKey(keyPair)
        val encrypt = RsaUtil.encrypt(src, publicKey)
        val decrypt = RsaUtil.decrypt(encrypt, privateKey)
        decrypt eq src
    }
}